### PR TITLE
executor: fix mppIterator memory leak when got error

### LINKED ISSUE #52406: mppIterator memory leak when got error
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)
https://github.com/pingcap/tidb/blob/3e2f2c5873eae1f5f2e4191f0f2064260e3b86ab/pkg/executor/internal/mpp/local_mpp_coordinator.go#L499-L510

when got error in L500, `stream` will not close, which may cause memory leak. Check:
image

The following is goroutine stack(grep `minutes`, you can see many stacks stuck on grpc `CallbackSerializer`)
[goroutine.txt](https://github.com/pingcap/tidb/files/14902250/goroutine.txt)

workload: the same workload of https://github.com/tidbcloud/tidb-cse/issues/938

tidb log: [tidb.log.tar.gz](https://github.com/pingcap/tidb/files/14914958/copy.tidb.log.tar.gz)

### 4. What is your TiDB version? (Required)
v8.0.0

### PR DESCRIPTION
**Pull request by guo-shaoge**
### What problem does this PR solve?

Issue Number: close #52406

Problem Summary: check issue for detail description

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [x] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: pkg/executor/internal/mpp/local_mpp_coordinator.go [hunk 1]
```diff
@@ -498,6 +498,7 @@ func (c *localMppCoordinator) cancelMppTasks() {
 func (c *localMppCoordinator) receiveResults(req *kv.MPPDispatchRequest, taskMeta *mpp.TaskMeta, bo *backoff.Backoffer) {
 	stream, err := c.sessionCtx.GetMPPClient().EstablishMPPConns(kv.EstablishMPPConnsParam{Ctx: bo.GetCtx(), Req: req, TaskMeta: taskMeta})
 	if err != nil {
+		stream.Close()
```
Comment (windtalker) [inline pkg/executor/internal/mpp/local_mpp_coordinator.go:501]: How about move L511 after L499?

Comment (guo-shaoge) [reply pkg/executor/internal/mpp/local_mpp_coordinator.go:501]: Golang lint error: `pkg/executor/internal/mpp/local_mpp_coordinator.go:500:2: should check returned error before deferring stream.Close() (SA5001)`

**Review by windtalker — APPROVED** [review]
LGTM

**Review by SeaRise — APPROVED** [review]

**Pull request merged**